blockchain/state_manager/src/utils.rs

```diff
@@ -104,3 +105,116 @@ where
         Ok(out)
     }
 }
+
+pub fn is_valid_for_sending(network_version: NetworkVersion, actor: &ActorState) -> bool {
+    // Comments from Lotus:
+    // Before nv18 (Hygge), we only supported built-in account actors as senders.
+    //
+    // Note: this gate is probably superfluous, since:
+    // 1. Placeholder actors cannot be created before nv18.
+    // 2. EthAccount actors cannot be created before nv18.
+    // 3. Delegated addresses cannot be created before nv18.
+    //
+    // But it's a safeguard.
+    //
+    // Note 2: ad-hoc checks for network versions like this across the codebase
+    // will be problematic with networks with diverging version lineages
+    // (e.g. Hyperspace). We need to revisit this strategy entirely.
+    if network_version < NetworkVersion::V18 {
+        return is_account_actor(&actor.code);
+    }
+
```

Comment:
I would keep this Lotus comment as well here.

`// After nv18, we also support other kinds of senders.`